src/magic_folder/web.py

```diff
@@ -19,6 +21,115 @@
 )
 
 
+def magic_folder_resource(get_magic_folder, get_auth_token, _v1_resource=None):
+    """
+    Create the root resource for the Magic Folder HTTP API.
+
+    :param get_magic_folder: See ``magic_folder_web_service``.
+    :param get_auth_token: See ``magic_folder_web_service``.
+
+    :param IResource _v1_resource: A resource which will take the place of the
+        usual bearer-token-authorized `/v1` resource.  This is intended to
+        make testing easier.
+
+    :return IResource: The resource that is the root of the HTTP API.
+    """
+    if _v1_resource is None:
+        _v1_resource = V1MagicFolderAPI(get_magic_folder)
+
+    root = Resource()
+    root.putChild(
+        b"api",
+        MagicFolderWebApi(get_magic_folder, get_auth_token),
+    )
+    root.putChild(
+        b"v1",
+        BearerTokenAuthorization(
+            _v1_resource,
+            get_auth_token,
+        ),
+    )
+    return root
```

Comment:
This looks like the root will have two resources side-by-side (`/v1` and `/api`) .. I would assume the intent is to have `/v1/api` with `api` a child of `BearerTokenAuthorization` instead of root?

Reply:
It will.  `/api` is all of the old API stuff though.  I just tried to leave it in place.  I expect we'll delete it once enough of the new API is done.  Actually, "enough" doesn't need to be very much since the only thing the old API exposes is that buggy "status" endpoint I think.  Anyway, the intent here was "leave `/api` as it is, add a new `/v1` which requires `Authorization`-header-based authorization for itself and all children" and later on we'll hang more and more stuff off of `/v1`.

Reply:
Ah, okay.